app/src/main/java/com/sun/colornotetaking/ui/main/HomeFragment.java

```diff
@@ -0,0 +1,155 @@
+    private TextView mTextPinedTaskTitle;
+    private HomeContract.Presenter mPresenter;
+    private TaskAdapter mPinedAdapter;
+    private TaskAdapter mOtherAdapter;
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        initView(view);
+        //initPresenter
+        TaskDAO taskDAO = TaskDAOImpl.getInstance(getContext(), SubItemDAOImpl.getInstance(getContext()));
+        TaskDataSource taskDataSource = TaskLocalDataSource.getInstance(taskDAO);
+        TaskRepository taskRepository = TaskRepository.getInstance(taskDataSource);
+        setPresenter(new HomePresenter(taskRepository, this));
+        if (mPresenter != null) mPresenter.start();
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_main_layout, container, false);
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setHasOptionsMenu(true);
+    }
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        menu.clear();
+        inflater.inflate(R.menu.menu_toolbar_main_item, menu);
+
+        //event search
+        MenuItem menuItem = menu.findItem(R.id.menu_search);
+        SearchView searchView = (SearchView) menuItem.getActionView();
+        searchView.setImeOptions(EditorInfo.IME_ACTION_DONE);
+        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String s) {
+                return false;
+            }
+
+            @Override
+            public boolean onQueryTextChange(String s) {
+                mPinedAdapter.getFilter().filter(s);
+                mOtherAdapter.getFilter().filter(s);
+                return true;
+            }
+        });
+        super.onCreateOptionsMenu(menu, inflater);
+    }
+
+    @Override
+    public void showPinedTasks(List<Task> PinedTasks) {
+        if (PinedTasks.size() != 0) mTextPinedTaskTitle.setVisibility(View.VISIBLE);
+        mPinedAdapter = new TaskAdapter(getContext(), PinedTasks);
+        mPinedTaskRecyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
+        mPinedTaskRecyclerView.setAdapter(mPinedAdapter);
+    }
+
+    @Override
+    public void showOtherTasks(List<Task> OtherTasks) {
+        if (OtherTasks.size() != 0) mTextOtherTaskTitle.setVisibility(View.VISIBLE);
+        mOtherAdapter = new TaskAdapter(getContext(), OtherTasks);
+        mNormalTaskRecyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
+        mNormalTaskRecyclerView.setAdapter(mOtherAdapter);
+    }
+
+    @Override
+    public void showGetDataError(Exception e) {
+        if (e == null)
```

Comment:
thiếu mở đóng khối lệnh
```suggestion
        if (e == null) {
```